packages/firestore/ios/RNFBFirestore/RNFBFirestoreModule.m

```diff
@@ -124,9 +123,10 @@ + (BOOL)requiresMainQueueSetup {
     : (RCTPromiseRejectBlock)reject
 ) {
     FIRFirestore *instance = [RNFBFirestoreCommon getFirestoreForApp:firebaseApp];
-    [instanceCache removeObjectForKey: [firebaseApp name]];
     
     [instance terminateWithCompletion:^(NSError *error) {
+      [instanceCache removeObjectForKey: [firebaseApp name]];
```

Comment:
This should be in the `else`? Otherwise, if it errors it should hasn't terminated but is still removed.